src/segment/segmentBase.js

```diff
@@ -55,3 +56,58 @@ export const segmentsFromBase = (attributes) => {
 
   return [segment];
 };
+
+export const addSegmentsToPlaylist = (playlist, sidx, baseUrl) => {
+  // Retain init segment information
+  const initSegment = playlist.sidx.map ? playlist.sidx.map : null;
```

Comment:
Would it make sense to store most of this in an object, something like `segmentDefaults` and then merge it below with attributes, that way we don't have to declare a lot of the same things twice?

Reply:
not sure I follow your meaning wrt declaring things twice

Reply:
I guess declaring things twice isn't really what I meant. This is what I mean:

```js
// current vars that are used as default attributes of 
// segments
const defaultAttributes = {
  baseUrl,
  // Retain source timeline
  timescale: sidx.timescale,
  timeline: playlist.timeline || 0,
  periodIndex: playlist.timeline || 0,
  // Retain source duration from initial master manifest parsing
  sourceDuration: playlist.sidx.duration,
};

...
// then in the loop below where we declare attributes now
const attributes = merge(defaultAttributes, {
  // duration of the referenced (sub)segment, in  the  timescale
  // this will be converted to seconds when generating segments
  duration: reference.subsegmentDuration, 
  indexRange: `${startIndex}-${endIndex}`
};
```

Reply:
This is probably better, but unless there are major objections, I'd prefer not to do that change right now.

Reply:
ok